lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -491,31 +520,47 @@ export class UserAgentApplication {
         scopes = this.filterScopes(scopes);
       }
 
-      // Extract ADAL id_token if it exists
-      let idTokenObject;
-      idTokenObject = this.extractADALIdToken();
+      // construct extraQueryParams string from the request
+      let extraQueryParameters = Utils.constructExtraQueryParametersString(request.extraQueryParameters);
+      extraQueryParameters = Utils.addPromptParameter(extraQueryParameters, request.prompt);
+      let user = this.getUser();
 
-      // silent login if ADAL id_token is retrieved successfully - SSO
-      if (idTokenObject && !scopes) {
-        this.logger.info("ADAL's idToken exists. Extracting login information from ADAL's idToken ");
-        extraQueryParameters = Utils.constructUnifiedCacheExtraQueryParameter(idTokenObject, extraQueryParameters);
+      // if the developer provides one of these, give preference to developer choice
+      if (request.account || request.sid || request.loginHint) {
+        extraQueryParameters = Utils.constructUnifiedCacheExtraQueryParameter(extraQueryParameters, request, null);
```

Comment:
all of this extraQueryParam construction could be factored out so the application logic just reads

const constructedQueryParams = Utils.constructQueryParams(req) 

the control flow doesn't care about the logic that goes into appending stuff

Reply:
Optimized in the new code.